Server/controllers/leave.controller.js

```diff
@@ -1,18 +1,58 @@
 "use strict";
+const leavesDataLayer = require('../datalayer/leave.datalayer');
 /**
  * Class representing leaves controller
  *
  * @class leavesController
  */
 class leavesController {
 
-    static getLeavesByUserId (req, res) {}
+    static async getLeavesByUserId (req, res) {
+        const userId = req.query.user;
+        let leavesList;
+        try {
+            leavesList = await leavesDataLayer.getLeavesByUserId(userId);
```

Comment:
Before this line we should check if `userId` exists, if not then return with 'Bad request'

Reply:
Check for undefined added.